flexinput/src/main/java/com/lytefast/flexinput/fragment/CameraFragment.kt

```diff
@@ -224,32 +208,47 @@ open class CameraFragment : PermissionsFragment() {
   }
 
 
-  private fun onCameraFacingClick(facingBtn: ImageView) {
-    val currentFlashState = cameraView.facing
+  private fun onCameraFacingClick() {
+    val currentFlashState = cameraView?.facing
     val currentStateIndex = FACING_STATE_CYCLE_LIST.indices.firstOrNull {
       currentFlashState == FACING_STATE_CYCLE_LIST[it]
     } ?: 0
 
     val newStateIndex = (currentStateIndex + 1) % FACING_STATE_CYCLE_LIST.size
-    setFacing(facingBtn, FACING_STATE_CYCLE_LIST[newStateIndex])
+    setFacing(FACING_STATE_CYCLE_LIST[newStateIndex])
   }
 
-  private fun onTakePhotoClick() {
-    cameraView.apply {
-      if (isCameraOpened) {
-        try {
-          takePicture()
-        } catch (e: Exception) {
-          onCameraError(e, "Camera error on take picture")
-          Toast.makeText(context, R.string.camera_unknown_error, Toast.LENGTH_SHORT).show()
+  @UiThread
+  private fun CameraKitView.onTakePhotoClick() {
+    captureImage { cameraKitView, imageData->
+      Log.d(TAG, "onPictureTaken ${imageData?.size}")
+      if (imageData == null) return@captureImage
+
+      val context = cameraKitView.context?: return@captureImage
+      Toast.makeText(context, "Picture saved", Toast.LENGTH_SHORT).show()
```

Comment:
Not localized toast. Anyway we can move this to a strings file, or allowed to be passed in?

Reply:
Just removed this since I don't think it makes sense. We are updating the externalAttachments to add the file, and most apps save these photos anyways